### DISCUSSION TITLE
Storybook Angular Issue with React errors

### DISCUSSION THREAD

**Discussion by yasminlopes** [storybookjs/storybook#24733] [Help] [answered]
My project uses Storybook with Angular in version 16, configured with a monorepository using NX. However, when opened on port 4400, Storybook displays a blank page and gives React errors.

![image](https://github.com/storybookjs/storybook/assets/75736181/41277360-ae2a-4cb8-a83b-967721b6b1ab)

![image](https://github.com/storybookjs/storybook/assets/75736181/afead13f-7f6c-4085-9551-371a35577b7e)

**main.ts**:
`import type { StorybookConfig } from '@storybook/angular';

const config: StorybookConfig = {
  stories: ['../**/*.stories.@(js|jsx|ts|tsx|mdx)'],
  addons: [
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
    '@storybook/addon-docs',
  ],
  framework: {
    name: '@storybook/angular',
    options: {},
  },
  docs: {
    autodocs: true
  }
};

export default config;`

**tsconfig.base.json: where is included my path**
`"paths": {
      "@app-ui": ["../lib/app-ui/src/index.ts"],`

**Package json**:

`
`{
  "name": "app",
  "version": "4.4.3",
  "scripts": {
    "start-dev": "nx serve",
    "build-prod": "node --max_old_space_size=10240 node_modules/@angular/cli/bin/ng build --configuration=production",
    "build-testing": "node --max_old_space_size=10240 node_modules/@angular/cli/bin/ng build --configuration=testing",
    "build": "gulp sass && nx build --configuration production",
    "ng": "ng",
    "test": "nx test",
    "start-build": "nx build",
    "start-dev-consumidor": "nx serve modules-consumidor",
    "start-dev-fornecedor": "nx serve modules-fornecedor",
    "start-dev-atendente": "nx serve modules-atendente",
    "storybook": "nx run lib-app-ui:storybook",
    "prepare": "husky install",
    "lint": "lint-staged"
  },
  "engines": {
    "node": "v16.14.0",
    "npm": "8.3.1"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^16.2.0",
    "@angular/cdk": "^16.2.1",
    "@angular/cdk-experimental": "^16.2.1",
    "@angular/common": "^16.2.0",
    "@angular/compiler": "^16.2.0",
    "@angular/core": "^16.2.0",
    "@angular/forms": "^16.2.0",
    "@angular/http": "^7.2.15",
    "@angular/material": "^16.2.1",
    "@angular/platform-browser": "^16.2.0",
    "@angular/platform-browser-dynamic": "^16.2.0",
    "@angular/platform-server": "^16.2.1",
    "@angular/pwa": "^16.2.0",
    "@angular/router": "^16.2.0",
    "@angular/service-worker": "^16.2.1",
    "@fnando/cnpj": "^1.0.2",
    "@fnando/cpf": "^1.0.2",
    "@ng-select/ng-option-highlight": "^11.1.1",
    "@ng-select/ng-select": "^11.1.1",
    "@ngneat/until-destroy": "^10.0.0",
    "@swimlane/ngx-datatable": "^20.1.0",
    "angular-cropperjs": "^14.0.1",
    "angular-dropdown": "^0.5.5",
    "arrive": "^2.4.1",
    "bootstrap": "^5.3.1",
    "bootstrap-notify": "^3.1.3",
    "bootstrap-select": "^1.13.18",
    "bootstrap-switch": "^3.4.0",
    "bootstrap-table": "^1.22.1",
    "bootstrap-tagsinput": "^0.7.1",
    "browser-filesaver": "^1.1.1",
    "buffer": "^6.0.3",
    "chart.js": "^4.3.3",
    "chartist": "^1.3.0",
    "chartist-plugin-barlabels": "^0.0.5",
    "chartist-plugin-htmllabels": "^0.4.0",
    "chartist-plugin-legend": "^0.6.2",
    "chartist-plugin-tooltips": "^0.0.17",
    "chartist-plugin-tooltips-updated": "^1.0.0",
    "chartist-plugin-zoom": "^0.6.0",
    "chartjs-plugin-zoom": "^2.0.1",
    "clipboard": "^2.0.11",
    "core-js": "^3.32.0",
    "create-file-list": "^1.1.0",
    "cropperjs": "^1.5.13",
    "datatables": "^1.10.18",
    "datatables.net-bs": "^1.13.6",
    "datatables.net-responsive": "^2.5.0",
    "date-fns": "^2.30.0",
    "draggable": "^4.2.0",
    "easy-pie-chart": "^2.1.7",
    "eonasdan-bootstrap-datetimepicker": "^4.17.47",
    "express": "^4.18.2",
    "file-saver": "^2.0.5",
    "font-awesome": "^4.7.0",
    "fullcalendar": "^5.8.0",
    "html2canvas": "^1.4.1",
    "intro.js": "^7.2.0",
    "jasny-bootstrap": "^4.0.0",
    "jquery": "^3.7.0",
    "jspdf": "^2.5.1",
    "jspdf-autotable": "^3.5.31",
    "jszip": "^3.10.1",
    "lazysizes": "^5.3.2",
    "lodash": "^4.17.21",
    "mime": "^3.0.0",
    "mime-wrapper": "^2.0.0",
    "moment": "^2.29.4",
    "ng-recaptcha": "^12.0.2",
    "ng-wormhole": "^3.0.2",
    "ng2-file-upload": "^5.0.0",
    "ngx-bootstrap": "^11.0.2",
    "ngx-color-picker": "^14.0.0",
    "ngx-currency": "^4.0.0",
    "ngx-drag-drop": "^16.0.0",
    "ngx-file-drop": "^16.0.0",
    "ngx-mask": "^16.2.9",
    "node-sass": "^9.0.0",
    "nouislider": "^9.2.0",
    "platform": "^1.3.6",
    "powerbi-client-angular": "^3.0.5",
    "quill": "^1.3.7",
    "rxjs": "~7.8.0",
    "sass": "^1.66.0",
    "socket.io-client": "^4.7.2",
    "tslib": "^2.3.0",
    "twitter-bootstrap-wizard": "^1.2.0",
    "underscore": "^1.13.6",
    "url": "^0.11.1",
    "validator": "^13.11.0",
    "vis-data": "^7.1.6",
    "vis-timeline": "^7.7.2",
    "xregexp": "^5.1.1",
    "zone.js": "~0.13.0"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^16.2.0",
    "@angular-devkit/core": "~16.2.0",
    "@angular-devkit/schematics": "~16.2.0",
    "@angular-eslint/eslint-plugin": "~16.0.0",
    "@angular-eslint/eslint-plugin-template": "~16.0.0",
    "@angular-eslint/template-parser": "~16.0.0",
    "@angular/cli": "~16.2.0",
    "@angular/compiler-cli": "^16.2.0",
    "@angular/language-service": "^16.2.0",
    "@nx/angular": "16.7.2",
    "@nx/eslint-plugin": "^16.7.3",
    "@nx/jest": "16.7.2",
    "@nx/js": "16.7.2",
    "@nx/storybook": "16.7.2",
    "@nx/web": "16.7.2",
    "@nx/workspace": "16.7.2",
    "@schematics/angular": "~16.2.0",
    "@storybook/addon-docs": "^7.4.0",
    "@storybook/addon-essentials": "7.3.2",
    "@storybook/addon-interactions": "^7.2.1",
    "@storybook/angular": "^7.3.2",
    "@storybook/builder-webpack5": "^7.3.2",
    "@storybook/core-server": "7.3.2",
    "@storybook/jest": "~0.1.0",
    "@storybook/manager-api": "^7.5.2",
    "@storybook/test-runner": "^0.13.0",
    "@storybook/testing-library": "~0.2.0",
    "@storybook/theming": "^7.5.2",
    "@types/bootstrap": "^5.2.6",
    "@types/chartist": "^1.0.0",
    "@types/jasmine": "~4.3.0",
    "@types/jest": "^29.4.0",
    "@types/jquery": "^3.5.16",
    "@types/lodash": "^4.14.197",
    "@types/node": "^20.5.0",
    "@types/xregexp": "^4.4.0",
    "@typescript-eslint/eslint-plugin": "^6.7.4",
    "@typescript-eslint/parser": "^5.62.0",
    "codelyzer": "^6.0.2",
    "concurrently": "^8.2.0",
    "eslint": "^8.51.0",
    "eslint-config-prettier": "^8.10.0",
    "eslint-plugin-import": "^2.28.1",
    "eslint-plugin-prettier": "^5.0.0",
    "git-commit-msg-linter": "^5.0.4",
    "gulp": "^4.0.2",
    "gulp-sass": "^5.1.0",
    "husky": "^8.0.0",
    "jasmine-core": "~4.6.0",
    "jasmine-spec-reporter": "^7.0.0",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "jest-preset-angular": "~13.1.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.2.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.1.0",
    "lint-staged": "^14.0.1",
    "nx": "16.7.2",
    "nx-cloud": "latest",
    "prettier": "^2.8.8",
    "prettier-eslint": "^15.0.1",
    "ts-jest": "^29.1.0",
    "ts-node": "^10.9.1",
    "typescript": "~5.1.3"
  }
}
``

**Comment by valentinpalkovic** [comment]
Hi @yasminlopes

Thank you for reporting this. Can you provide a reproduction?

**Reply by yasminlopes** [reply]
Hi @valentinpalkovic 

To reproduce the issue, 
- you need to have Angular v16.2.0 installed
- local library with Nx called `lib-app-ui`. which is where Storybook is configured.
- run `npm run storybook` or `nx run lib-app-ui:storybook`.

my folder structure look like this:
![image](https://github.com/storybookjs/storybook/assets/75736181/2f1c6819-c2e2-417e-af23-5f629c8915de)

**Comment by valentinpalkovic** [comment]
@mandarini Does the issue sound familiar to you?

**Reply by mandarini** [reply]
Thanks, I'll try to take a look tomorrow!

**Comment by mandarini** [comment]
Hello @yasminlopes ! Unfortunately I could not reproduce your issue. Here is my repository: 
[https://github.com/mandarini/nx-16-7-2-storybook](https://github.com/mandarini/nx-16-7-2-storybook)

Can you please provide a simple reproduction repository that I could look into? Or can you compare your setup with the repository I shared? I am using the versions you are using!

**Comment by mandarini** [comment]
Oh, by the way, I do not see the following dependencies in your `package.json`:

```
   "react": "^18.2.0",
    "react-dom": "^18.2.0",
```

React is needed to run Storybook. Please correct me if I'm wrong @valentinpalkovic !

**Reply by valentinpalkovic** [reply]
Exactly! Currently, React is a required peer dependency of Storybook

**Reply by yasminlopes** [reply]
Maybe that's the issue, the missing react dependency. I'll take a look later, but doesn't that increase the package size?

**Comment by yasminlopes** [comment]
I installed the dependencies of React, but the problem persists :( 
I tried to reproduce the error in a new repository, but it worked perfectly. Maybe it is something specific to this project… I will investigate, because it worked correctly and stopped suddenly. Maybe some change caused this… I appreciate your help for now 🙏

![image](https://github.com/storybookjs/storybook/assets/75736181/3ab54d79-3956-4472-94db-902d719d5da8)

**Reply by yasminlopes** [reply, marked as answer]
During my investigation, all of my attempts resulted in failures. 😐

Therefore, I decided to remove Storybook from the project. Since it only had the configuration defined, there would be no problem in reinstalling it. I followed the reinstallation method that @mandarini provided in their [repository ](https://github.com/mandarini/nx-16-7-2-storybook) and, to my satisfaction, it worked perfectly 🎉

Although the original problem remains a mystery, I am relieved that everything is now working correctly. I appreciate the help received @valentinpalkovic @mandarini 

![image](https://github.com/storybookjs/storybook/assets/75736181/9d81cf75-12a6-4f5c-9259-ba0a0770e283)

**Reply by mandarini** [reply]
That's great to hear @yasminlopes ! :D DO let us know if you need any more support!

**Reply by yasminlopes** [reply]
Thank you.

**Comment by mandarini** [comment]
I'll go ahead and close this as resolved!
